src/Cache/InMemoryDriver.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Tnapf\Config\Cache;
+
+use DateInterval;
+use Psr\SimpleCache\CacheInterface;
+use Tnapf\Config\Exceptions\InvalidCacheKeyException;
+
+class InMemoryDriver implements CacheInterface
+{
+    private array $storage = [];
+
+    public function get(string $key, mixed $default = null): mixed
+    {
+        return $this->has($key) ? $this->storage[$key] : $default;
+    }
+
+    public function set(string $key, mixed $value, DateInterval|int|null $ttl = null): bool
+    {
+        $this->validateKey($key);
+        $this->storage[$key] = $value;
+
+        return true;
+    }
+
+    public function delete(string $key): bool
+    {
+        $this->validateKey($key);
+        unset($this->storage[$key]);
+
+        return true;
+    }
+
+    public function clear(): bool
+    {
+        $this->storage = [];
+
+        return true;
+    }
+
+    public function getMultiple(iterable $keys, mixed $default = null): iterable
+    {
+        $items = [];
+        foreach ($keys as $key) {
+            $items[$key] = $this->storage[$key] ?? $default;
+        }
+
+        return $items;
+    }
+
+    public function setMultiple(iterable $values, DateInterval|int|null $ttl = null): bool
+    {
+        foreach ($values as $key => $value) {
+            $this->set($key, $value, $ttl);
+        }
+
+        return true;
+    }
+
+    public function deleteMultiple(iterable $keys): bool
+    {
+        foreach ($keys as $key) {
+            $this->delete($key);
+        }
+
+        return true;
+    }
+
+    public function has(string $key): bool
+    {
+        $this->validateKey($key);
+
+        return isset($this->storage[$key]) || array_key_exists($key, $this->storage);
+    }
+
+    private function validateKey(string $key): void
```

Comment:
To my understanding, PSR requires you to support `A-Z, a-z, 0-9, _, .` but allows implementations to support more.

There's no technical reason I can think of any reason to block these characters explicitly. Am I missing something?

Reply:
Assuming there's a reason for not allowing these characters, please add `@throws` doc blocks where relevant

Reply:
> To my understanding, PSR requires you to support `A-Z, a-z, 0-9, _, .` but allows implementations to support more.

From [PSR-16](https://www.php-fig.org/psr/psr-16/#12-definitions):
> The following characters are reserved for future extensions and MUST NOT be supported by implementing libraries: {}()/\@: